operation/fhir-operation-validate/src/test/java/com/ibm/fhir/operation/validate/ValidateOperationTest.java

```diff
@@ -268,4 +284,218 @@ public void testProfileParmProfileUnsupportedError() {
+        Patient patient = (Patient) getTestResource("Patient.json");
+        FHIRResourceHelpers resourceHelper = mock(FHIRResourceHelpers.class);
+        SingleResourceResult result = mock(SingleResourceResult.class);
+
+     // mock and return a resource when resourceHelper.doRead() is invoked from validateOperation.doInvoke
+        when(result.isSuccess()).thenReturn(true);
+        when(result.getResource()).thenReturn(patient);
+        when(resourceHelper.doRead(eq("Patient"), anyString())).thenAnswer(x -> result);
+        
+        FHIROperationContext operationContext =
+                FHIROperationContext.createInstanceOperationContext("validate");
+        Parameters input = Parameters.builder()
+                .build();
+        
+        Issue expectedOutput = Issue.builder()
+                    .severity(IssueSeverity.INFORMATION)
+                    .code(IssueType.INFORMATIONAL)
+                    .details(CodeableConcept.builder()
+                        .text(string("All OK"))
+                        .build())
+                    .build();
+        Parameters output = validateOperation.doInvoke(operationContext, Patient.class, "1", null, input, resourceHelper, null);
+        OperationOutcome operationOutcome = output.getParameter().get(0).getResource().as(OperationOutcome.class);
+        assertEquals(operationOutcome.getIssue().size(), 1);
+        assertEquals(operationOutcome.getIssue().get(0), expectedOutput);
+    }
+    
+    /**
+     * Test resource-instance level validate operation when the resource with the input logicalID is not available in the database.
+     * @throws Exception 
+     */
+    @Test(expectedExceptions = { FHIROperationException.class } , expectedExceptionsMessageRegExp  = ".*Patient with id '1' was not found*")
+    public void testResourceInstanceLevelValidateForNullResource() throws Exception {
+        Patient patient = (Patient) getTestResource("Patient.json");
+        FHIRResourceHelpers resourceHelper = mock(FHIRResourceHelpers.class);
+        SingleResourceResult result = mock(SingleResourceResult.class);
+        
+        // mock and return null when resourceHelper.doRead() is invoked from validateOperation.doInvoke
+        when(result.isSuccess()).thenReturn(false);
+        when(result.getResource()).thenReturn(null);
+        when(resourceHelper.doRead(eq("Patient"), anyString())).thenAnswer(x -> result);
+        
+        FHIROperationContext operationContext =
+                FHIROperationContext.createInstanceOperationContext("validate");
+        Parameters input = Parameters.builder()
+                .build();
+        
+        Issue expectedOutput = Issue.builder()
+                    .severity(IssueSeverity.INFORMATION)
+                    .code(IssueType.INFORMATIONAL)
+                    .details(CodeableConcept.builder()
+                        .text(string("All OK"))
+                        .build())
+                    .build();
```

Comment:
we're expecting the exception in this case, so maybe leave this out and end with the doInvoke call?

Reply:
this has been fixed. Cleaned up unwanted code.